pandas/core/frame.py

```diff
@@ -2864,6 +2865,74 @@ def to_parquet(
             **kwargs,
         )
 
+    def to_orc(
+        self,
+        path: FilePath | WriteBuffer[bytes] | None = None,
+        engine: Literal["pyarrow"] = "pyarrow",
+        index: bool | None = None,
+        **kwargs,
+    ) -> bytes | None:
```

Comment:
Since this is a new function, lets already only allow keyword-only arguments (except for path; makes it easier to add overloads):

```py
@overload
def to_orc(self, path: None, *, engine:Literal["pyarrow"]=..., ...) -> bytes:
    # replace default parameters in overloads with ... (but you stil need to add the other keyword-arguments, I just skipped them)
    ...

@overload
def to_orc(self, path: FilePath | WriteBuffer[bytes], *, engine:Literal["pyarrow"]=..., ...) -> None:
    ...

@overload
def to_orc(self, path: FilePath | WriteBuffer[bytes] | None, *, engine:Literal["pyarrow"]=..., ...) -> bytes | None:
    ...

def to_orc(self, path: FilePath | WriteBuffer[bytes] | None, *, engine:Literal[...]="pyarrow"..., ...) -> bytes | None:
    implementation
```

Reply:
Fixed!

Reply:
Did you push your changes?